test/clojure/core_test/sort_by.cljc

```diff
@@ -0,0 +1,151 @@
+          '() nil
+          '(1 2 3 4) [3 1 2 4]
+          '(nil 2 3 4) [3 nil 2 4]
+          '(1 2 3 4) '(3 1 2 4)
+          '(1 2 3 4) #{3 1 2 4}
+          '([:a 1] [:b 2] [:c 3]) {:c 3 :b 2 :a 1}
+          '([1] [2] [3] [4]) [[3] [1] [2] [4]]
+          '("a" "b" "c" "d") ["b" "a" "c" "d"]
+          '(\c \e \j \l \o \r \u) "clojure"))
+      (testing "key-fn is keyword"
+        (is (= (seq [{:a 4 :b 7}
+                     {:a 3 :b 8}
+                     {:a 2 :b 9}
+                     {:a 1 :b 10}])
+               (sort-by :b simple-vec-maps)))
+        (is (= (seq [{:a 1 :b 10}
+                     {:a 2 :b 9}
+                     {:a 3 :b 8}
+                     {:a 4 :b 7}])
+               (sort-by :a simple-vec-maps)))
+        ;; test more complex key-fn and stability of sort-by
+        ;; :a + :b = 11 for all elements of simple-vec-maps
+        (is (= (seq simple-vec-maps)
+               (sort-by #(+ (:a %) (:b %)) simple-vec-maps)))
+        ;; If the key-fn returns the same value all the time we also
+        ;; expect no change. For instance, when the key-fn returns
+        ;; `nil` constantly because we use the wrong keyword or we
+        ;; navigate the structure of the element incorrectly.
+        (is (= (seq simple-vec-maps)
+               (sort-by :c simple-vec-maps)))))
+    
+    (testing "sort-by artity-3"
+      ;; Use `compare`
+      (is (= (seq [{:a 1 :b 10}
+                   {:a 2 :b 9}
+                   {:a 3 :b 8}
+                   {:a 4 :b 7}])
+             (sort-by :a compare simple-vec-maps)))
+      ;; Reverse `compare`
+      (is (= (seq [{:a 4 :b 7}
+                   {:a 3 :b 8}
+                   {:a 2 :b 9}
+                   {:a 1 :b 10}])
+             (sort-by :a #(compare %2 %1) simple-vec-maps)))
+      (is (= (seq [{:a 1 :b 10}
+                   {:a 2 :b 9}
+                   {:a 3 :b 8}
+                   {:a 4 :b 7}])
+             (sort-by :a < simple-vec-maps)))
+      (is (= (seq [{:a 4 :b 7}
+                   {:a 3 :b 8}
+                   {:a 2 :b 9}
+                   {:a 1 :b 10}])
+             (sort-by :a > simple-vec-maps)))
+
+      ;; Use `compare`
+      (is (= (seq [{:a 4 :b 7}
+                   {:a 3 :b 8}
+                   {:a 2 :b 9}
+                   {:a 1 :b 10}])
+             (sort-by :b compare simple-vec-maps)))
+      ;; Reverse `compare`
+      (is (= (seq [{:a 1 :b 10}
+                   {:a 2 :b 9}
+                   {:a 3 :b 8}
+                   {:a 4 :b 7}])
+             (sort-by :b #(compare %2 %1) simple-vec-maps)))
+      (is (= (seq [{:a 4 :b 7}
+                   {:a 3 :b 8}
+                   {:a 2 :b 9}
+                   {:a 1 :b 10}])
+             (sort-by :b < simple-vec-maps)))
+      (is (= (seq [{:a 1 :b 10}
+                   {:a 2 :b 9}
+                   {:a 3 :b 8}
+                   {:a 4 :b 7}])
+             (sort-by :b > simple-vec-maps))))
```

Comment:
Are these testing any unique functionality that the compare + reverse compare cases don't cover? Looks to me like over-testing which we can remove.